File: utilcode/pkg/src/main/java/com/blankj/utilcode/pkg/feature/bar/BarActivity.java
package com.blankj.utilcode.pkg.feature.bar;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import com.blankj.lib.base.BaseBackActivity;
import com.blankj.utilcode.pkg.R;

/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2017/05/27
 *     desc  : demo about BarUtils
 * </pre>
 */
public class BarActivity extends BaseBackActivity {

    private TextView tvAboutStatus;

    public static void start(Context context) {
        Intent starter = new Intent(context, BarActivity.class);
        context.startActivity(starter);
    }

    @Override
    public void initData(@Nullable Bundle bundle) {

    }

    @Override
    public int bindLayout() {
        return R.layout.activity_bar;
    }

    @Override
    public void initView(Bundle savedInstanceState, View contentView) {
        setTitle(R.string.demo_bar);

        tvAboutStatus = findViewById(R.id.tv_about_status);
        findViewById(R.id.btn_status_bar).setOnClickListener(this);
        findViewById(R.id.btn_status_bar_color).setOnClickListener(this);
        findViewById(R.id.btn_status_bar_alpha).setOnClickListener(this);
        findViewById(R.id.btn_status_bar_image_view).setOnClickListener(this);
        findViewById(R.id.btn_status_bar_fragment).setOnClickListener(this);
        findViewById(R.id.btn_status_bar_swipe_back).setOnClickListener(this);
        findViewById(R.id.btn_status_bar_drawer).setOnClickListener(this);
        findViewById(R.id.btn_notification_bar).setOnClickListener(this);
        findViewById(R.id.btn_nav_bar).setOnClickListener(this);
    }

    @Override
    public void doBusiness() {

    }

    @Override
    public void onWidgetClick(View view) {
        int i = view.getId();
        if (i == R.id.btn_status_bar) {
            BarStatusActivity.start(this);

        } else if (i == R.id.btn_status_bar_color) {
            BarStatusColorActivity.start(this);

        } else if (i == R.id.btn_status_bar_alpha) {
            BarStatusAlphaActivity.start(this);

        } else if (i == R.id.btn_status_bar_image_view) {
            BarStatusImageViewActivity.start(this);

        } else if (i == R.id.btn_status_bar_fragment) {
            BarStatusFragmentActivity.start(this);

        } else if (i == R.id.btn_status_bar_swipe_back) {
            BarStatusSwipeBackActivity.start(this);

        } else if (i == R.id.btn_status_bar_drawer) {
            BarStatusDrawerActivity.start(this);

        } else if (i == R.id.btn_notification_bar) {
            BarNotificationActivity.start(this);

        } else if (i == R.id.btn_nav_bar) {
            BarNavActivity.start(this);

        }
    }
}

File: utilcode/pkg/src/main/java/com/blankj/utilcode/pkg/feature/bar/BarActivity.kt
package com.blankj.utilcode.pkg.feature.bar

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import com.blankj.lib.base.BaseTitleBarActivity
import com.blankj.utilcode.pkg.R
import kotlinx.android.synthetic.main.activity_bar.*

/**
 * ```
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 2017/05/27
 * desc  : demo about BarUtils
 * ```
 */
class BarActivity : BaseTitleBarActivity() {

    companion object {
        fun start(context: Context) {
            val starter = Intent(context, BarActivity::class.java)
            context.startActivity(starter)
        }
    }

    override fun bindTitle(): CharSequence {
        return getString(R.string.demo_bar)
    }

    override fun initData(bundle: Bundle?) {}

    override fun bindLayout(): Int {
        return R.layout.activity_bar
    }

    override fun initView(savedInstanceState: Bundle?, contentView: View) {
        barStatusAboutBtn.setOnClickListener(this)
        barStatusSetColorBtn.setOnClickListener(this)
        barStatusSetAlphaBtn.setOnClickListener(this)
        barStatusSetImageViewBtn.setOnClickListener(this)
        barStatusSetCustomBtn.setOnClickListener(this)
        barStatusSetFragmentBtn.setOnClickListener(this)
        barStatusSetSwipeBackBtn.setOnClickListener(this)
        barStatusSetDrawerBtn.setOnClickListener(this)
        barNotificationAboutBtn.setOnClickListener(this)
        barNavAboutBtn.setOnClickListener(this)
    }

    override fun doBusiness() {}

    override fun onWidgetClick(view: View) {
        when (view.id) {
            R.id.barStatusAboutBtn -> BarStatusActivity.start(this)
            R.id.barStatusSetColorBtn -> BarStatusColorActivity.start(this)
            R.id.barStatusSetAlphaBtn -> BarStatusAlphaActivity.start(this)
            R.id.barStatusSetImageViewBtn -> BarStatusImageViewActivity.start(this)
            R.id.barStatusSetCustomBtn -> BarStatusCustomActivity.start(this)
            R.id.barStatusSetFragmentBtn -> BarStatusFragmentActivity.start(this)
            R.id.barStatusSetSwipeBackBtn -> BarStatusSwipeBackActivity.start(this)
            R.id.barStatusSetDrawerBtn -> BarStatusDrawerActivity.start(this)
            R.id.barNotificationAboutBtn -> BarNotificationActivity.start(this)
            R.id.barNavAboutBtn -> BarNavActivity.start(this)
        }
    }
}
